binsync/decompilers/binja/binja_binsync.py

```diff
@@ -176,6 +176,7 @@ def symbol_updated(self, view, sym):
 
     def type_defined(self, view, name, type_):
         l.debug(f"Type Defined: {name} {type_}")
+        name = str(name)
```

Comment:
This is the only _needed_ fix, everything else is versioning and making debug print quiet

Reply:
I'm kinda curious now: How did this lead to the situation in #300?

Reply:
@alexrp hehe yeah it's a fun story, so it does something like this:
1. After BS sees the change it sends it through the pipeline to be to be set into the state. It does this by dumping many `Enum` artifacts here:
https://github.com/binsync/binsync/blob/28bdacc7420b644e4126ac15c0c4e0c973c651fa/binsync/data/state.py#L282

The `Enum.dump_many(self.enums)` function call there returns a Dict, where right now the key is this special `QualifiedName` Binja object that still prints as a string. 
So we are doing something like this right now:
```py
toml.dumps({QualifiedName("MY_ENUM"): {"name: QualifiedName("MY_ENUM"), "member": ...}})
```

Where things go bad is in the `toml.dumps` function call, which attempts to guess types! For the key, it MUST be a string, so TOML always converts the thing into a str. It happens to work out. For the value of the dict keys, it will try more introspection to use native types. We trigger this branch in toml lib:
https://github.com/uiri/toml/blob/5706d3155f4da8f3b84875f80bfe0dfc6565f61f/toml/encoder.py#L177

Why? Because this C++ to Python converted object has a `__iter__` attr, but is not a Native Python type that TOML can lookup here:
https://github.com/uiri/toml/blob/5706d3155f4da8f3b84875f80bfe0dfc6565f61f/toml/encoder.py#L134

And that's how it became a List in the value but not in the Key :). Aren't encodings fun hahahahhaha.